### DISCUSSION TITLE
Dependency Resolution Algorithm

### DISCUSSION THREAD

**Discussion by macovedj** [bytecodealliance/registry#163] [General]
As we are able to generate a tree of nested dependencies, we will need to settle on an algorithm that will be used for handling multiple versions of the same package specified across various dependencies (which version(s) will we grab if multiple versions are specified by different packages).  We can discuss this here.

**Comment by lann** [comment]
I think "[Minimal Version Selection](https://research.swtch.com/vgo-mvs)" (which Go uses) is a pretty decent default as long as components obey semver. In practice you still need some mechanism to override versions, though I suspect that would apply to any resolution algorithm.
